### PR TITLE
Add nonkube site commands and unit tests

### PR DESCRIPTION
**Pull request by lynnemorrison**
Add update, delete and status site CLI commands
Currently create router-access if user configures enable-link-access.
Add unit tests.

### CODE REVIEW

File: internal/cmd/skupper/site/nonkube/site_delete.go [hunk 1]
```diff
@@ -2,29 +2,82 @@ package nonkube
...
+	} else {
+		ok, err := resourceStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("site name is not valid: %s", err))
+		} else {
+			cmd.siteName = args[0]
+		}
+	}
+
+	return validationErrors
```
Comment (fgiorgetti) [inline internal/cmd/skupper/site/nonkube/site_delete.go:62]: No error is returned when deleting an invalid site name.

File: internal/cmd/skupper/site/nonkube/site_update.go [hunk 2]
```diff
@@ -2,33 +2,236 @@ package nonkube
...
+		if site == nil || err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("site %s must exist in namespace %s to be updated", cmd.siteName, cmd.namespace))
+		} else {
+			// save existing values
+
+		}
+
+		routerAccessName := "router-access-" + cmd.siteName
+		routerAccess, err := cmd.routerAccessHandler.Update(routerAccessName)
+		if err == nil && routerAccess != nil {
```
[the remark this answers is not in the document]
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_update.go:94]: @fgiorgetti I was trying to allow a way the user could remove routerAccess with an update command.
If with an updated command the user doesn't set the enableLinkAccess then routerAccess is removed.  

Do you think it is better to leave routerAccess if user had already configured it with the create command?
Once configured with the create or an update command the only way to delete routerAccess would be then for the user to delete the site.

File: internal/cmd/skupper/site/nonkube/site_create.go [hunk 3]
```diff
@@ -146,7 +134,7 @@ func (cmd *CmdSiteCreate) Run() error {
 		},
 		Spec: v2alpha1.SiteSpec{
 			Settings:   cmd.options,
-			LinkAccess: cmd.linkAccessType,
+			LinkAccess: "default",
```
[the remark this answers is not in the document]
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_create.go]: Should be removed.

File: internal/cmd/skupper/site/nonkube/site_delete.go (hunk 1 above)
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_delete.go:70]: added warning

File: internal/cmd/skupper/site/nonkube/site_update.go [hunk 4]
```diff
@@ -2,33 +2,236 @@ package nonkube
...
+			validationErrors = append(validationErrors, fmt.Errorf("bindhost is not valid: a valid IP address or hostname is expected"))
+		} else {
+			cmd.newSettings.bindHost = cmd.Flags.BindHost
+		}
+	}
+	if cmd.Flags != nil && len(cmd.Flags.SubjectAlternativeNames) != 0 {
+		for _, name := range cmd.Flags.SubjectAlternativeNames {
+			ip := net.ParseIP(name)
+			ok, _ := hostStringValidator.Evaluate(name)
+			if !ok || ip == nil {
```
[the remark this answers is not in the document]
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_update.go]: fixed

File: internal/cmd/skupper/site/nonkube/site_update.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fgiorgetti) [reply internal/cmd/skupper/site/nonkube/site_update.go:94]: When I look at the update command for the connector resource, I noticed that I do not need to provide all the flags. So if I omit the routing-key on an update, the previous one is preserved.

I was expecting the same behavior here. If the `--enable-link-access` was set during site crete, but not passed (not changed) during an update, then the original value is preserved (enable-link-access=true).

So what seems to be missing here is to load the original state of the enable-link-access field based on the current input state. If user wants to disable it, they can provide `--enable-link-access=false`. Wdyt?

File: internal/cmd/skupper/site/nonkube/site_create.go [hunk 5]
```diff
@@ -77,29 +69,34 @@ func (cmd *CmdSiteCreate) ValidateInput(args []string) []error {
...
 		if !ok {
-			validationErrors = append(validationErrors, fmt.Errorf("link access type is not valid: %s", err))
+			validationErrors = append(validationErrors, fmt.Errorf("output type is not valid: %s", err))
 		}
 	}
 
-	if !cmd.Flags.EnableLinkAccess && len(cmd.Flags.LinkAccessType) > 0 {
-		validationErrors = append(validationErrors, fmt.Errorf("for the site to work with this type of linkAccess, the --enable-link-access option must be set to true"))
+	if cmd.Flags != nil && cmd.Flags.BindHost == "" && cmd.Flags.EnableLinkAccess {
+		validationErrors = append(validationErrors, fmt.Errorf("bindhost should not be empty"))
```
Comment (fgiorgetti) [inline internal/cmd/skupper/site/nonkube/site_create.go:82]: I am wondering why not set `0.0.0.0` as the default value for bind host?
It is currently required, but if enable-link-access is not true, it is not used for anything.

File: internal/nonkube/client/fs/site_handler.go [hunk 6]
```diff
@@ -34,6 +38,79 @@ func (s *SiteHandler) Add(resource v2alpha1.Site) error {
...
+	var context v2alpha1.Site
+	fileName := name + ".yaml"
+
+	if opts.RuntimeFirst == true {
+		// First read from runtime directory, where output is found after bootstrap
+		// has run.  If no runtime sites try and display configured sites
+		err, file := s.ReadFile(s.pathProvider.GetRuntimeNamespace(), fileName, common.Sites)
+		if err != nil {
+			if opts.LogWarning {
+				fmt.Println("Site not initialized yet")
```
Comment (fgiorgetti) [inline internal/nonkube/client/fs/site_handler.go:51]: I believe we should write this warning message to stderr.
The reason is that if you do `skupper site status my-site-name -o yaml > site.yaml` the warning message is included into the output file.

File: internal/nonkube/client/fs/site_handler.go [hunk 7]
```diff
@@ -34,6 +38,79 @@ func (s *SiteHandler) Add(resource v2alpha1.Site) error {
...
+
+func (s *SiteHandler) List() ([]*v2alpha1.Site, error) {
+	var sites []*v2alpha1.Site
+
+	// First read from runtime directory, where output is found after bootstrap
+	// has run.  If no runtime sites try and display configured sites
+	path := s.pathProvider.GetRuntimeNamespace()
+	err, files := s.ReadDir(path, common.Sites)
+	if err != nil {
+		fmt.Println("Site not initialized yet")
```
Comment (fgiorgetti) [inline internal/nonkube/client/fs/site_handler.go:96]: Same thing here, maybe better writing it to stderr.

File: internal/cmd/skupper/site/nonkube/site_update.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fgiorgetti) [reply internal/cmd/skupper/site/nonkube/site_update.go:94]: Now the router access is being preserved or removed properly, but I noticed that when I run the update
command the "linkAccess: default" is being added, no matter what.

File: internal/nonkube/client/fs/site_handler.go (hunk 6 above)
Comment (lynnemorrison) [reply internal/nonkube/client/fs/site_handler.go]: changed to stderr

File: internal/nonkube/client/fs/site_handler.go (hunk 7 above)
Comment (lynnemorrison) [reply internal/nonkube/client/fs/site_handler.go]: fixed

File: internal/cmd/skupper/site/nonkube/site_create.go (hunk 5 above)
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_create.go]: defaulted bind-host

File: internal/cmd/skupper/site/nonkube/site_update.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_update.go:94]: accidently left linkAccess in the update.   It is no removed.

File: internal/cmd/skupper/site/nonkube/site_create.go [hunk 8]
```diff
@@ -48,12 +46,10 @@ func (cmd *CmdSiteCreate) NewClient(cobraCommand *cobra.Command, args []string)
 }
 
 func (cmd *CmdSiteCreate) ValidateInput(args []string) []error {
-
 	var validationErrors []error
-
-	if cmd.Flags.ServiceAccount != "" {
```
Comment (nluaces) [inline internal/cmd/skupper/site/nonkube/site_create.go:54]: Why is this warning deleted?

**Review by nluaces — APPROVED** [review]

File: internal/cmd/skupper/site/nonkube/site_create.go (hunk 8 above)
Comment (lynnemorrison) [reply internal/cmd/skupper/site/nonkube/site_create.go:54]: looks like a mistake.   I'll add it back

**Review by nluaces — APPROVED** [review]

**Review by fgiorgetti — APPROVED** [review]
LGTM

**Pull request merged**